xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -61,6 +66,10 @@
 // TODO(chengyuanzhang): put data types into smaller categories.
 final class EnvoyProtoData {
   static final String TRANSPORT_SOCKET_NAME_TLS = "envoy.transport_sockets.tls";
+  static final String HTTP_FAULT_FILTER_NAME = "envoy.fault";
+  @VisibleForTesting
+  static boolean enableFaultInjection =
```

Comment:
Why define the env var here? You can alway parse configurations regardless the feature is turned on or not. It should be used to gate the feature implementation (in XdsNameResolver).

Reply:
I don't think we should couple response parsing with specific features. We should just parse whatever data we want from response messages based on our development stage, but not based on what features we are releasing. A invalid configuration is always invalid. I believe most importantly we are using env var to gate our implementation codepath, because we have not tested that and it could be buggy/wrong. We do not want code to execute branches implementing the feature. Ideally, it should gate both name resolver's code and response parsing code. But, the parsing code is more mechanical and less critical, plus we aren't applying any of our own feature-specific logic to the parsing. The fault inject configurations are either absent or being valid.

> group all env variables to a class XdsEnv

That sounds a horrible idea, at least for Java (like our existing XdsLbPolicies, it's horrible Java). Those env vars aren't related to other, we shouldn't just dump everything together just because we want easy references.

Reply:
Parse proto regardless of feature flag now.